### ISSUE TITLE
Clarification on the nomination workflow and the status field

### ISSUE THREAD

**Issue by jiripetrzelka** [erasmus-without-paper/ewp-specs-api-imobilities#9]
I have a couple of questions regarding the flow chart from MBR on nominations, which I paste below with two arrows added at the top of the chart):

<img width="709" alt="image" src="https://github.com/erasmus-without-paper/ewp-specs-api-imobilities/assets/1330161/5c2ccbee-677f-4240-9b6f-95116dce4d93">

1. How is the sending institution supposed to distinguish the "Pending: sent but not yet delivered" phase from the "Pending: delivered" phase?
Am I right that the first phase corresponds to an empty imobilities response for the given omobility-id, while the latter state corresponds to a non-empty response with the "status" element set to "pending"?

2. I assume the "Verified: accepted" phase corresponds to the "verified" status and the "Verified: rejected" phase corresponds to the "rejected" status. I suggest to rename the "verified" status to "accepted".

3. Look at the orange arrow I added to the chart. Am I right that once the receving institution rejected the nomination, it should start detecting any change in any of the omobilities fields and assume a transition directly to the "Pending: delivered" phase once any change is detected (usually by means of omobilities CNR)?

4. Look at the green arrow I added to the chart. A nomination has been accepted but a change in omobilities response is detected. 
Examples of changes:
- last name (due to marriage).
- iia-id change (due to mistake).
- status changed to cancelled.
- sending-academic-term-ewp-id changed due to extension.

In this case, do we transition to the "Pending: delivered" phase and does the coordinator at the receving institution have to verify the nomination again, or is the sending institution required to make some fields in the omobilities response immutable once the nomination has been accepted? If so, what is the set of immutable fields whose change should be considered an error and reported to the Stats Portal?

**Comment by janinamincer-daszkiewicz** [comment]
> How is the sending institution supposed to distinguish the "Pending: sent but not yet delivered" phase from the "Pending: delivered" phase?

The easiest approach is to look at the response to CNR: 200 means that nomination is Pending but delivered.

> Am I right that the first phase corresponds to an empty imobilities response for the given omobility-id, while the latter state corresponds to a non-empty response with the "status" element set to "pending"?

This approach is better, but requires more effort from the sending HEI. I would leave it to the local system whether they implement it in an easier way or more elaborated.

> I assume the "Verified: accepted" phase corresponds to the "verified" status and the "Verified: rejected" phase corresponds to the "rejected" status. 

Yes

> I suggest to rename the "verified" status to "accepted".

What is the opinion of the others?

> Look at the orange arrow ...

Yes 
> Look at the green arrow ...

Yes, with the exception of 'status changed to cancelled' - this one does not need confirmation (verification) by the receiving HEI (see https://github.com/erasmus-without-paper/ewp-specs-api-omobilities).

**Comment by jiripetrzelka** [comment]
Thank you for the answers, Janina.

Concerning the status field, my intention was to have the naming in line with the MBR. However, we should also think about the division of the nomination and the application phase here. Currently, the verified status is bound only to the nomination being "formally correct with the cooperation conditions":
https://github.com/erasmus-without-paper/ewp-specs-api-imobilities/blob/a2ec90f80b08028ff8778282ce3ec8fae04b633f/endpoints/get-response.xsd#L158-L167

But even if the student's nomination is "formally" correct, the mobility can later be refused during the application process, for example because the receiving institution finds out some insufficiency in student's academic performance. Should the receiving institution set the status to "refused" in this case? I suppose it should but if the result of the application process is be communicated through this API, it would be more consistent if the successful application process would also be communicated. Status "verified" would then correspond to the formal verification of the nomination and a new status "accepted" would mean that not only is the nomination formally valid but the student has been successful in the application phase as well.

**Comment by janinamincer-daszkiewicz** [comment]
> Should the receiving institution set the status to "refused" in this case?

The receiving institution should set state to Rejected, see MBR, page 8, Step 5: "_In rare cases, the application is rejected if the student does not meet the enrolment qualifications_".

> if the result of the application process is be communicated through this API, it would be more consistent if the successful application process would also be communicated. 

Such step is not foreseen in MBR. If the status verified is not later replaced by Rejected, it stays verified (accepted).

I will ask RMs for confirmation.

**Comment by jiripetrzelka** [comment]
> The receiving institution should set state to Rejected, see MBR, page 8, Step 5: "_In rare cases, the application is rejected if the student does not meet the enrolment qualifications_".

Then I suggest to be more clear about it in the specification and change the annotation of this element to "The nomination or the application has been rejected by the receiving HEI."
https://github.com/erasmus-without-paper/ewp-specs-api-imobilities/blob/38914aac38ec9d98fbbc441bfbad5943ec4bed76/endpoints/get-response.xsd#L168-L174


> Such step is not foreseen in MBR. If the status verified is not later replaced by Rejected, it stays verified (accepted).

Ok, but using this logic we should remove statuses "live" and "recognized" in omobilities because these are processes that are connected to the application phase, not the nomination phase. Or am I mistaken and the MBR describes the "live" and "recognized" phases somewhere?

**Comment by janinamincer-daszkiewicz** [comment]
> Or am I mistaken and the MBR describes the "live" and "recognized" phases somewhere?

You are not mistaken, however these statuses already exist in the specification. The idea was to make the smallest set of necessary changes to the specification.

**Comment by janinamincer-daszkiewicz** [comment]
During the IF meeting on 2024-09-18 Relationship managers shared with us a feedback from BPOs. I copy here this part which is relevant to the questions from this thread.

> Status clarification in case of changes (which are rather very rare cases and mostly regards change of mobility period): 
> 
> - Changes should be possible only until the nomination is pending verification of the receiving HEI
>   - After any updates, sending HEI needs to send CNR.
>   - Notification needs to be clearly signalled to the receiving HEI.
> - In later stages, any changes can be introduced by cancelling the current nomination and re-nominating the student.
> - These changes are exceptional and, to ensure clarity, should be discussed between the HEIs via other channels first.

**Comment by janinamincer-daszkiewicz** [comment]
This part from BPOs feedback is also relevant to this thread.

> Nominations and extension of stay: 
> 
> - The nomination does not need to be updated if the stay is extended within the same academic year (with some very rare exceptions that require new nomination, e.g. mobility financed by different grants).
> - The extensions that span over two academic years should be treated as two separate mobilities, and in such cases new nomination should be sent to the host institution.

**Comment by janinamincer-daszkiewicz** [comment]
The most recent feedback from BPOs is shared in https://github.com/erasmus-without-paper/ewp-specs-api-omobilities/pull/56#issuecomment-2676093215.

**Comment by janinamincer-daszkiewicz** [comment]
According to this feedback neither green nor orange arrows in https://github.com/erasmus-without-paper/ewp-specs-api-imobilities/issues/9#issue-2310146737 are allowed.
